### DISCUSSION TITLE
How to pass cli argument from Clap into Rockets config

### DISCUSSION THREAD

**Discussion by wimvelzeboer** [rwf2/Rocket#2632] [Questions]
Hi,

I am using Clap to capture cli arguments, something like the port which Rocket need to use.
```
myapi --port 9090
```
In my clap routing I collect the desired port number, and from there I call the `main` method to launch the Rocket.
I would like to pass the port number as parameter into the main method but that is not possible.

What would be the best approach to combine Clap with Rocket, where Clap can be used to collect all sorts of configuration settings to be used in Rocket.

And another thing, I only want to launch Rocket when certain cli arguments are used, so not in all cases. So that I can do something like:
```bash
# initialise database, which will create all required tables
myapi init  --database-url=postgresql://username:password@localhost:5432/dbname
# start api daemon
myapi start --port=9090
```

**Comment by SergioBenitez** [comment]
See [custom providers](https://rocket.rs/v0.5-rc/guide/configuration/#custom-providers).
